### PR TITLE
fix: Improve UI for the BlocksDrawer with long labels

### LINKED ISSUE #2343: New blocks drawer: Block names are cut-off
# Bug Report

Arc 2023-03-17 at 21 30 19@2x

On certain screen sizes (which happened to be the size of my macbook+browser, rip), the block names in the new block drawer are cut-off. Those should always be readable, especially for sensible names (my names aren't too long)

**Payload version:** 1.6.22

### PR DESCRIPTION
**Pull request by ChrisGV04**
## Description

Fixes #2343 

I made the `BlocksDrawer` fit 6 blocks per row on XL screens to fit a bit more of the labels in it and implemented the `title` HTML property on the `ThumbnailCard` so that even when the title gets cut by the `overflow: hidden; text-overflow: ellipsis;` combo, if the user hovers over the card, every browser shows a small tooltip with the full label, which was the issue mentioned by @AlessioGr 

Screenshot 2023-03-18 at 10 14 03 p m

- [x] I have read and understand the CONTRIBUTING.md document in this repository

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Checklist:

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Existing test suite passes locally with my changes
- [ ] I have made corresponding changes to the documentation

### CODE REVIEW

File: src/admin/components/elements/ThumbnailCard/index.tsx
Comment (jacobsfletch) [inline src/admin/components/elements/ThumbnailCard/index.tsx:36]: We need to avoid using nested ternaries. Ultimately this should be an ESLint rule, but that's work for another day.

**Review by jacobsfletch — CHANGES_REQUESTED** [review]
We need to avoid using nested ternaries but also standardize the label itself so that the `title` and the rendered label are the same. Something like this:

```ts
let label = labelFromProps;
if (!label) label = typeof doc?.filename === 'string' ? doc?.filename : `[${t('untitled')}]`}
```

**Comment by jacobsfletch** [comment]
@ChrisGV04 this is great! I left some very small feedback here mostly around code style, so if you can make those few changes we should be good to merge this one in 🙌

**Comment by ChrisGV04** [comment]
Hey @jacobsfletch sure thing!

**Comment by ChrisGV04** [comment]
@jacobsfletch It seems that the master branch and my PR are now in a conflict because the master branch now uses the new title functionality for dot notation titles. So I believe someone from the core team should add the title attribute to the `ThumbnailCard` wrapper that I implemented on this PR to the master branch

**Comment by jacobsfletch** [comment]
Ahh, yea. You could rebase this PR to resolve the conflicts yourself, but I'm now realizing that https://github.com/payloadcms/payload/pull/2350 overlaps with this work as well. So that will need to get merged and then implemented into this component first and it may be easiest to open a fresh PR for this.

**Comment by ChrisGV04** [comment]
That's right. So I think I'll wait for that PR to be merged for me to rebase and create a new PR with this changes to avoid any conflicts

**Comment by jacobsfletch** [comment]
@ChrisGV04 all set here! Go ahead and re-open this PR if you are able to get it rebased or open a new PR which might be easier.

**Comment by ChrisGV04** [comment]
On it! I'm opening a new PR in a few minutes

**Pull request closed without merging**
